src/common/transformations/src/transformations/op_conversions/einsum_decomposition.cpp

```diff
@@ -278,15 +474,22 @@ ov::Output<ov::Node> reshape_input_for_matmul(const ov::Output<ov::Node>& input_
     } else if (separate_sub_shape.size() > 0) {
         // in this case compute a product of separate dimension sizes since they must be
         // presented with just one dimension for MatMul
-        auto reduce_axis_const = ov::op::v0::Constant::create(ov::element::Type_t::i64, ov::Shape{1}, {0});
         auto separate_shape_prod =
             std::make_shared<ov::op::v1::ReduceProd>(separate_sub_shape[0], reduce_axis_const, true);
         separate_parts.push_back(separate_shape_prod->output(0));
         subgraph_nodes.insert(subgraph_nodes.end(), {reduce_axis_const, separate_shape_prod});
     }
+    ov::OutputVector reduced_sub_shape_prod;
+    auto const_0 = ov::op::v0::Constant::create(ov::element::i32, {1}, {0});
+    if (reduced_sub_shape.size() > 0) {
+        auto product = std::make_shared<ov::op::v1::ReduceProd>(reduced_sub_shape[0], const_0, true);
+        subgraph_nodes.insert(subgraph_nodes.end(), {reduce_axis_const, product});
```

Comment:
Sometimes the Constant ops are added to the subgraph nodes vector, sometimes not. In fact the latest approach to track the added nodes is usage of NodeRegistry, but I understand that when introducing changes into existing transformation, it can be easier to follow the approach already used in the rest of the file. 

Also there is a chance that ``reduce_axis_const`` will be insterted twice (L480 and L486), just a warning, maybe it's not an issue.

```suggestion
        subgraph_nodes.insert(subgraph_nodes.end(), {reduce_axis_const, const_0, product});
```